vm/src/vm.rs

```diff
@@ -187,10 +187,15 @@ fn map_cow_option<T, U, F>(cow: Cow<T>, f: F) -> Option<Cow<U>>
 
 impl VMEnv {
     pub fn find_type_info(&self, name: &str) -> Result<Cow<types::Alias<Symbol, TcType>>> {
-        if let Some(alias) = self.type_infos.id_to_type.get(name) {
-            return Ok(Cow::Borrowed(alias));
-        }
         let name = Name::new(name);
+        let module_str = name.module().as_str();
+        if module_str == "" {
+            return self.type_infos
+                .id_to_type
+                .get(name.as_str())
+                .map(|alias| Cow::Borrowed(alias))
+                .ok_or_else(|| Error::UndefinedBinding(name.as_str().into()));
+        }
```

Comment:
type_infos contain globally defined types so if `module_str == ""` the input is just an identifier (`Int` not `std.prelude.Option`) and the only way to return a result is a global type. I should probably handle the case of a bare global variable though (`test.test` works but `test` wont)